client/api/go-client/client_test.go

```diff
@@ -52,6 +53,59 @@ func setupHeketiServer(app *glusterfs.App) *httptest.Server {
 	return httptest.NewServer(n)
 }
 
+func TestTopology(t *testing.T) {
+	db := tests.Tempfile()
+	defer os.Remove(db)
+
+	// Create the app
+	app := glusterfs.NewTestApp(db)
+	defer app.Close()
+
+	// Setup the server
+	ts := setupHeketiServer(app)
+	defer ts.Close()
+
+	// Create cluster correctly
+	c := NewClient(ts.URL, "admin", TEST_ADMIN_KEY)
+	tests.Assert(t, c != nil)
+	cluster, err := c.ClusterCreate()
+	tests.Assert(t, err == nil)
+	tests.Assert(t, cluster.Id != "")
+	tests.Assert(t, len(cluster.Nodes) == 0)
+	tests.Assert(t, len(cluster.Volumes) == 0)
+
+	// Get information about the client
+	info, err := c.ClusterInfo(cluster.Id)
+	tests.Assert(t, err == nil)
+	tests.Assert(t, reflect.DeepEqual(info, cluster))
+
+	// Get information about the Topology and verify the cluster creation
+	topology, err := c.TopologyInfo()
+	tests.Assert(t, err == nil)
+	tests.Assert(t, topology.ClusterList[0].Id == cluster.Id)
+
+	// Create the nodereq
+	nodeReq := &glusterfs.NodeAddRequest{}
+	nodeReq.ClusterId = cluster.Id
+	nodeReq.Hostnames.Manage = []string{"manage"}
+	nodeReq.Hostnames.Storage = []string{"storage"}
+	nodeReq.Zone = 10
+
+	//Add node to the existing cluster
+	node, err := c.NodeAdd(nodeReq)
+	tests.Assert(t, err == nil)
+	tests.Assert(t, node.Zone == nodeReq.Zone)
+	tests.Assert(t, node.Id != "")
+	tests.Assert(t, reflect.DeepEqual(nodeReq.Hostnames, node.Hostnames))
+	tests.Assert(t, len(node.DevicesInfo) == 0)
+
+	//// Get information about the Topology and verify the node add
+	topology, err = c.TopologyInfo()
+	tests.Assert(t, err == nil)
+	tests.Assert(t, topology.ClusterList[0].Nodes[0].Id == node.Id)
+
```

Comment:
You probably want to add more nodes and devices and volumes to correctly test the topology info.  You also would want to test it with multiple clusters.